packages/grid/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -122,9 +122,9 @@ export function useGridDimensions(
       };
 
       const scrollInformation = hasScroll({
-        content: { width: columnsTotalWidth, height: rowsMeta.currentPageTotalHeight },
+        content: { width: Math.round(columnsTotalWidth), height: rowsMeta.currentPageTotalHeight },
```

Comment:
In https://github.com/mui/mui-x/pull/9516 I didn't anticipate the floating point precision error 😅
Rather than removing `Math.round`, it would make more sense to use it on both sides of the equation (content width and container width)